theia-extensions/viewer-prototype/src/browser/trace-viewer/trace-viewer-contribution.ts

```diff
@@ -71,15 +73,16 @@ export class TraceViewerContribution
         };
     }
 
-    protected async launchTraceServer(rootPath?: string): Promise<void> {
+    protected async launchTraceServer(rootPath?: string, type?: TraceResourceType): Promise<void> {
         let healthResponse;
         try {
             healthResponse = await this.tspClient.checkHealth();
         } catch (err) {
             // continue to start trace server
         }
+        const selectFiles = type && type === TraceResourceType.FILE ? true : false;
         if (healthResponse && healthResponse.isOk() && healthResponse.getModel()?.status === 'UP') {
-            this.openDialog(rootPath);
+            this.openDialog(rootPath, selectFiles);
```

Comment:
I think we should change the title of the dialog. Right now the title is `Open Trace` for both cases. It might not be clear for the user. Maybe change the title to `Open Trace File`/`Open Trace Folder` or `Select Trace File to Open`/`Select Trace Folder to Open`.